### ISSUE TITLE
Upgrade to grahql-java 15.0

### ISSUE THREAD

**Issue by marceloverdijk** [leangen/graphql-spqr#356]
`graphql-spqr` is not yet compatible with latest `graphql-java` `15.0`.

```
java.lang.NoSuchMethodError: graphql/schema/GraphQLType.getName()Ljava/lang/String;
        at io.leangen.graphql.generator.TypeCache.<init>(TypeCache.java:19)
```

**Comment by cblanche34** [comment]
Same problem with graphql-java 14.0, getName doesn't exist anymore. 
Somebody know when 0.12 (compatible with 14.0 and maybe so on) will released ?

**Comment by hc-codersatlas** [comment]
@cblanche34 if you really need to use 14 jitpack works with it. The patch is in the source already. No work has been done for 15 so not sure.

**Comment by ankit-joinwal** [comment]
This also blocks using apollo federation with SPQR because we need to upgrade to graphql-java version 15 in order to exclude directives from being exported in schema. references https://github.com/leangen/graphql-spqr/issues/290

**Comment by ankit-joinwal** [comment]
@kaqqao any update on releasing 0.10.2-SNAPSHOT version as release?

**Comment by n614cd** [comment]
I also just ran into this issue. The GraphQL-Java Spring Boot MVC Starter supports either 12 or 15. Makes it rather interesting.

**Comment by jdariasl-adl** [comment]
Same problem here. cannot use latest version of micronaut and  micronaut-graphql because this issue.

**Comment by pan-daniel** [comment]
Also ran into this issue. https://github.com/graphql-java/graphql-java-spring webflux version 2.0 also requires GraphQL 15.

**Comment by dargenn** [comment]
Same problem using micronaut 2.0+ which has graphql as transitive dependency.

**Comment by kaqqao** [comment]
SPQR v0.11.0 shipped with graphql-java v16.1.
Major graphql-java releases are not backwards compatible, so never assume you can change the version that SPQR is using, even if it compiles successfully.
